dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/RenormalizationProcessor.java

```diff
@@ -0,0 +1,159 @@
+     * @see RenormalizationProcessor#renormalizeAllele(java.lang.String)
+     */
+    private boolean isAmbiguous(Variant variant) {
+        try {
+            boolean isIndel = variant.getType() == VariantType.INDEL;
+            boolean oneAlleleIsEmpty = variant.getReference().isEmpty() ^ variant.getAlternate().isEmpty();
+            return isIndel && oneAlleleIsEmpty && areContextAndLastNucleotideEqual(variant);
+        } catch (Exception e) {
+            // TODO jmmut: should we throw exception and stop the whole job if one variant is not found in the assembly? (or if there was another problem?)
+            logger.warn(e.getMessage(), e);
+            // if something went wrong with the fasta, we can not say it's ambiguous
+            return false;
+        }
+    }
+
+    private boolean areContextAndLastNucleotideEqual(Variant variant) {
+        String nonEmptyAllele = variant.getReference().isEmpty() ? variant.getAlternate() : variant.getReference();
+        char lastNucleotideInAllele = nonEmptyAllele.charAt(nonEmptyAllele.length() - 1);
+        char contextBaseInAssembly = getContextBaseInAssembly(variant);
+
+        return lastNucleotideInAllele == contextBaseInAssembly;
+    }
+
+    private char getContextBaseInAssembly(Variant variant) {
+        long contextPosition = variant.getStart() - 1;
+        String sequence = fastaSequenceReader.getSequence(variant.getChromosome(), contextPosition,
+                                                          contextPosition);
+        if (sequence == null || sequence.length() != 1) {
+            throw new RuntimeException(
+                    "Reference sequence could not be retrieved correctly for chromosome=\"" + variant.getChromosome()
+                            + "\", position=" + contextPosition);
+        }
+        char contextBaseInAssembly = sequence.charAt(0);
+        return contextBaseInAssembly;
+    }
+
+    /**
+     * Change the positions and the non empty allele to the desired representation
+     *
+     * For a complete explanation,
+     * @see RenormalizationProcessor#isAmbiguous(uk.ac.ebi.eva.commons.core.models.pipeline.Variant)
+     * @see RenormalizationProcessor#renormalizeAllele(java.lang.String)
+     */
+    private Variant renormalize(IVariant variant) {
+        String renormalizedReference = variant.getReference();
+        String renormalizedAlternate = variant.getAlternate();
+        long renormalizedStart = variant.getStart() - 1;
+        long renormalizedEnd = variant.getEnd() - 1;
+        if (variant.getReference().isEmpty()) {
+            renormalizedAlternate = renormalizeAllele(variant.getAlternate());
+        } else if (variant.getAlternate().isEmpty()) {
+            renormalizedReference = renormalizeAllele(variant.getReference());
+        } else {
+            throw new AssertionError("Can not re-normalize: not a standard indel: " + variant);
```

Comment:
Because we don't normalize before this step and we can't guarantee that dbSNP variants are fully normalized (most are, but well...) I'm a bit afraid of considering this as hard as an `AssertionError`.

But in case we decide to keep it or turn it into an exception, this could be a bit nicer to read:

```
Can not re-normalize due to non-standard INDEL:
```

Reply:
I put that as an assertion because the first part of this class (isAmbiguous) already checks for this, so this 3rd case should not happend unless we introduce a bug.

If the indel is not fully normalized, this class doesn't attempt to renormalize it